--- /Users/agrant/Desktop/folding.vim	2015-06-17 15:51:50.000000000 -0700
+++ /Users/agrant/.vim/ftplugin/latex-box/folding.vim	2015-06-17 15:52:04.000000000 -0700
@@ -256,6 +256,28 @@
     endif
 endfunction
 
+function! s:LabelNCCVuln()
+    let i = v:foldend
+    while i >= v:foldstart
+        if getline(i) =~ '^\s*\\vrisk'
+            return matchstr(getline(i), '^\s*\\vrisk{\zs.*\ze}')
+        end
+        let i -= 1
+    endwhile
+    return ""
+endfunction
+
+function! s:CaptionNCCVuln()
+    let i = v:foldend
+    while i >= v:foldstart
+        if getline(i) =~ '^\s*\\vtitle'
+            return matchstr(getline(i), '^\s*\\vtitle{\zs.\+\ze}')
+        end
+        let i -= 1
+    endwhile
+    return ""
+endfunction
+
 function! LatexBox_FoldText_title()
     let line = getline(v:foldstart)
     let title = 'Not defined'
@@ -303,6 +325,9 @@
         elseif env == 'table'
             let label = s:LabelEnv()
             let caption = s:CaptionTable()
+        elseif env == 'nccvuln'
+            let label = s:LabelNCCVuln()
+            let caption = s:CaptionNCCVuln()
         else
             let label = s:LabelEnv()
             let caption = s:CaptionEnv()
@@ -321,6 +346,8 @@
                         \ substitute(caption, '}\s*$', '',''))
         elseif caption == ''
             let title = printf('%-12s%56s', env, '(' . label . ')')
+        elseif env == 'nccvuln'
+            let title = printf('%s %s', '{' . label . '}', caption)
         else
             let title = printf('%-12s%-30s %21s', env . ':',
                         \ strpart(substitute(caption, '}\s*$', '',''),0,34),
@@ -334,7 +361,7 @@
 " {{{1 LatexBox_FoldText
 function! LatexBox_FoldText()
     let nlines = v:foldend - v:foldstart + 1
-    let title = strpart(LatexBox_FoldText_title(), 0, 68)
+    let title = LatexBox_FoldText_title()
     let level = ''
 
     " Fold level
